isisd/isis_nb_config.c

```diff
@@ -2862,6 +2865,12 @@ int isis_instance_flex_algo_create(struct nb_cb_create_args *args)
 			admin_group_allow_explicit_zero(
 				&fa->admin_group_include_all);
 		}
+		if (update_te) {
+			for (ALL_LIST_ELEMENTS_RO(area->circuit_list, node,
+						  circuit))
+				isis_link_params_update(circuit,
```

Comment:
Why do you not call `isis_link_params_update_asla()` directly? only flex_algo parameters have been updated and need attention to eventually regenerate the LSP.

Reply:
You are right. Changed
I have added the last flex-algo unconfiguration case and some cleanup